cmd/deploy.go

```diff
@@ -238,6 +238,20 @@ func (dc *deployCmd) loadAPIModel() error {
 				dc.containerService.Properties.CustomCloudProfile.IdentitySystem = dc.authProvider.getAuthArgs().IdentitySystem
 			}
 		}
+
+		if dc.containerService.Properties.IsAzureStackCloud() {
+			if dc.containerService.Properties.MasterProfile.Distro == api.AKSUbuntu1604 {
+				log.Errorln("Distro 'aks-ubuntu-16.04' is not longer supported on Azure Stack Hub, please update master profile distro in api model to use 'aks-ubuntu-18.04'")
+				return errors.New("Distro value 'aks-ubuntu-16.04' of master profile is not supported on Azure Stack Hub")
+			}
+
+			for _, app := range dc.containerService.Properties.AgentPoolProfiles {
```

Comment:
Could you please add unit tests to avoid regressions?

Reply:
Added